cli/fathom_web/test/test_test.py

```diff
@@ -0,0 +1,59 @@
+from click import BadParameter
+import pytest
+
+from ..commands.test import decode_weights
+
+
+def test_valid_json_input():
+    json_string = '{"coeffs": [["rule1", 0.1], ["rule2", 0.2]], "bias": 0.5}'
+    expected_dict = {
+        'coeffs': [
+            ['rule1', 0.1],
+            ['rule2', 0.2],
+        ],
+        'bias': 0.5,
+    }
+    decoded_weights = decode_weights(None, None, json_string)
+    assert decoded_weights == expected_dict
+
+
+def test_not_json():
+    run_invalid_json('not_json', r'.*valid.*')
+
+
+def run_invalid_json(json_string, assertion_match_regex):
+    with pytest.raises(BadParameter, match=assertion_match_regex):
```

Comment:
In general, if you want to import more deeply, I'm 100% onboard. For brevity's sake, I generally import as deeply as I can without becoming ambiguous.